include/grpc/grpc_security.h

```diff
@@ -191,6 +191,55 @@ int grpc_server_add_secure_http2_port(grpc_server *server, const char *addr,
 grpc_call_error grpc_call_set_credentials(grpc_call *call,
                                           grpc_credentials *creds);
 
+/* --- Authentication Context. --- */
+
+/* TODO(jboeuf): Define some well-known property names. */
+
+#define GRPC_TRANSPORT_SECURITY_TYPE_PROPERTY_NAME \
+  "transport_security_type"
+#define GRPC_FAKE_TRANSPORT_SECURITY_TYPE "fake"
+#define GRPC_SSL_TRANSPORT_SECURITY_TYPE "ssl"
+
+#define GRPC_X509_CN_PROPERTY_NAME "x509_common_name"
+#define GRPC_X509_SAN_PROPERTY_NAME "x509_subject_alternative_name"
+
+typedef struct grpc_auth_context grpc_auth_context;
+typedef struct grpc_auth_property_iterator grpc_auth_property_iterator;
+
+/* value, if not NULL, is guaranteed to be NULL terminated. */
+typedef struct grpc_auth_property {
+  char *name;
+  char *value;
+  size_t value_length;
+} grpc_auth_property;
+
+/* Returns NULL when the iterator is at the end. */
+const grpc_auth_property *grpc_auth_property_iterator_next(
+    grpc_auth_property_iterator *it);
+void grpc_auth_property_iterator_destroy(grpc_auth_property_iterator *it);
+
+grpc_auth_property_iterator *grpc_auth_context_property_iterator(
```

Comment:
This seems like it implies an allocation.
Would it be worth popping the iterator out here and providing an _init() function instead?